### ISSUE TITLE
SHACL `sh:class` constraint should test for `rdfs:subClassOf*` as well 

### ISSUE THREAD

**Issue by henrietteharmse** [eclipse-rdf4j/rdf4j#4513]
### Current Behavior

Current behaviour is that it only tests for `rdf:type`: 
EXISTS{?value a <http://purl.obolibrary.org/obo/ENVO_01000187>}

### Expected Behavior

Instead it should test against both `rdf:type` and `rdfs:subClassOf*`. According to the SHACL specification:

![image](https://user-images.githubusercontent.com/32327979/233341779-74b08af2-5937-496b-beb1-f4ca307ead86.png)

### Steps To Reproduce

GraphDB version 10.2.0

### Version

What ever is coming with GraphDB 10.2.0

### Are you interested in contributing a solution yourself?

Perhaps?

**Comment by henrietteharmse** [comment]
Ah! I understood the meaning of `rdf:type/rdfs:subClassOf` incorrectly. I was hoping there is a why to check subClassOf relations between _classes_. However, the SHACL specification does not address it.

**Comment by hmottestad** [comment]
I know that you closed this issue @henrietteharmse but you got me a bit curious what restriction you are trying to model. 

Any chance you can explain a bit more? Do you have any examples?

**Comment by henrietteharmse** [comment]
Hi Håvard,

Here is the typical use case which I suspect appears rather often:
1. You have an OWL2 ontology consisting of owl:classes.
2. You want to annotate your data with the classes from your ontology, restricting it to subclasses of some class in your ontology.
3. As the annotations are done using classes rather individuals, the `sh:class` restriction does not apply for this use case.

Here is a concrete example. There are different observatories for doing research related observation. Each observatory has 1 or more biomes. Therefore researchers want to annotate each observatory with classes that are subclasses of [ENVO_00000428](https://www.ebi.ac.uk/ols4/ontologies/envo/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FENVO_00000428?lang=en). 

My data looks as follows:

       :myObservatory rdf:type :Observatory;
          :hasBiomes obo:ENVO_00000447 . 

My shape data looks as follows:

     :ObservatoryShape
        rdf:type sh:NodeShape ;
        sh:targetClass :Observatory ;
        sh:property [
           sh:path :hasBiomes ;
           sh:minCount 1 ;
           sh:class obo:ENVO_00000428 ; #Here is issue
       ] .

Ideally because `obo:ENVO_00000447` is  a subclass of `obo:ENVO_00000428`,  the requirement is that the data validate successfully. However, the semantics of `sh:class` is:

    ASK {
      $value rdf:type/rdfs:subClassOf* $class .
    }

The mistake I made, is to think this is pseudo code meaning `rdf:type` OR `rdfs:subClassOf*` rather than proper SPARQL making use of the path sequence operator (`/`). Thus, on my database

    ASK {
      obo:ENVO_00000447 rdf:type/rdfs:subClassOf* obo:ENVO_00000428 .
    }

answers NO, while the following answers YES

    ASK {
       obo:ENVO_00000447 rdfs:subClassOf* obo:ENVO_00000428 .
    }

This is the use case I do not currently see any way to deal with in SHACL, or Shex for that matter. Hence, the reason why I closed the ticket here.

My current thinking is that SHACL (Shex) needs a `sh:subClassOf` constraint to deal with this use case.
